app/src/main/java/org/mozilla/focus/browser/cursor/CursorView.kt

```diff
@@ -38,11 +36,18 @@ class CursorView(context: Context, attrs: AttributeSet) : View(context, attrs) {
     private val radius = context.resources.getDimensionPixelSize(R.dimen.remote_cursor_size) / 2f
     private val paint = Paint().apply {
         style = Paint.Style.FILL
-        alpha = CURSOR_ALPHA
         isAntiAlias = true
-        shader = RadialGradient(radius, radius, /* radius */ radius,
-                ContextCompat.getColor(context, R.color.teal50), ContextCompat.getColor(context, R.color.photonBlue50),
-                Shader.TileMode.CLAMP)
+    }
+
+    private val cursorBg = BitmapFactory.decodeResource(resources, R.drawable.cursor_bg)
+    private val cursorBgBlendingMode = PorterDuffXfermode(PorterDuff.Mode.MULTIPLY)
+    private val cursorFg = BitmapFactory.decodeResource(resources, R.drawable.cursor_fg)
+    private val cursorFgBlendingMode = PorterDuffXfermode(PorterDuff.Mode.SCREEN)
+
+    init {
+        // In order to draw the bitmaps with the blending mode and
+        // alpha we have to set the LayerType to software
+        setLayerType(View.LAYER_TYPE_SOFTWARE, null)
```

Comment:
We probably don't want to do this: it's going to negatively affect page load performance if the Cursor needs to be redrawn (which I expect to be any time the content under the cursor is redrawn or the cursor is moved).

That is, unless you figured out that this is actually only drawn once and the GPU handles translations and redraws. In which case, please add a comment explaining this to be the case.

Do you disagree?

Reply:
@mcomella I agree, but I wasn't able to to keep the transparency without that. Any ideas for an alternative?

Reply:
@boek Not off the top of my head but if a UI effect hurts performance, especially pageload, I'd rather not do it.

Thoughts:
- The fennec platform team deals with low-level rendering: perhaps they have some ideas about how to do this? Talk to :snorp
- Have you asked UX if there is a similar, but simpler, implementation they could suggest?